rdr_service/genomic/genomic_job_components.py

```diff
@@ -677,8 +677,19 @@ def prep_aw2_row_attributes(self, row: dict, member: GenomicSetMember):
                 row['contamination'] = 0
 
         except ValueError:
-            logging.error(f'contamination must be a number for sample_id: {row["sampleid"]}')
-            return GenomicSubProcessResult.ERROR
+            if row['processingstatus'].lower() != 'pass':
+                return row
+
+            _message = f'contamination must be a number for sample_id: {row["sampleid"]}'
+
+            self.controller.create_incident(source_job_run_id=self.job_run_id,
+                                            source_file_processed_id=self.file_obj.id,
+                                            code=GenomicIncidentCode.DATA_VALIDATION_FAILED.name,
+                                            message=_message,
+                                            biobank_id=member.biobankId,
+                                            sample_id=row['sampleid'],
+                                            )
+
```

Comment:
If the ValueError occurs and the row's `processingstatus` is pass, then the code would continue to try to calculate the contamination category (below) and would raise another ValueError. I may be missing something, but I think this would cause a crash before it's done processing the file.

Reply:
You are right. We want to return here after `create_incident()` and then skip the row in the main ingestion loop. Since the incident is written to the incident table, there is a record of the erroneous row.